bin/instrument_server

```diff
@@ -8,443 +8,211 @@ rescue Gem::LoadError
-      output = { :gauges => {} }
-      if SystemInspector.command_present?('df', 'disk storage')
-        output[:gauges].merge!(disk_storage)
-      end
-      if SystemInspector.command_present?('mount', 'disk IO')
-        output[:gauges].merge!(disk_io)
-      end
-      output
-    end
+  opts.on('-t', '--temp-dir TEMP_DIRECTORY', "Where to store temporary files (default #{options[:tmp_location]})") do |t|
+    options[:tmp_location] = t
+  end
 
-    def self.disk_storage
-      output = {}
-      `df -Pka`.split("\n").each do |line|
-        device, total, used, available, capacity, mount = line.split(/\s+/)
-        if device == "tmpfs"
-          names = ["tmpfs_#{mount.gsub(/[^[:alnum:]]/, "_")}".gsub(/_+/, "_")]
-        elsif device =~ %r{/dev/}
-          names = [File.basename(device)]
-        else
-          next
-        end
-        names << 'root' if mount == '/'
-        names.each do |name|
-          output["disk.#{name}.total_mb"] = total.to_f / 1024
-          output["disk.#{name}.used_mb"] = used.to_f / 1024
-          output["disk.#{name}.available_mb"] = available.to_f / 1024
-          output["disk.#{name}.available_percent"] = available.to_f / total.to_f * 100
-        end
-      end
-      output
-    end
+  opts.on('--debug', "Print all sent metrics to the log") do
+    options[:debug] = true
+  end
 
-    def self.disk_io
-      mounted_devices = `mount`.split("\n").grep(/^\/dev\/(\w+)/) { $1 }
-      diskstats_lines = `cat /proc/diskstats`.split("\n").grep(/#{mounted_devices.join('|')}/)
-      entries = diskstats_lines.map do |line|
-        values = line.split
-        entry = {}
-        entry[:time] = Time.now
-        entry[:device] = values[2]
-        entry[:utilization] = values[12].to_f
-        SystemInspector.memory.store("disk_stats_#{entry[:device]}".to_sym, entry)
-      end
-
-      output = {}
-      entries.each do |entry|
-        if previous_entry = SystemInspector.memory.retrieve("disk_stats_#{entry[:device]}".to_sym)
-          time_delta = (entry[:time] - previous_entry[:time]) * 1000
-          utilization_delta = entry[:utilization] - previous_entry[:utilization]
-          output["disk.#{entry[:device]}.percent_utilization"] = utilization_delta / time_delta * 100
-        end
-      end
-      output
-    end
+  opts.on('-h', '--help', 'Display this screen') do
+    puts opts
+    exit
+  end
+end
 
-    def self.load_filesystem
-      output = { :gauges => {} }
-      if SystemInspector.command_present?('sysctl', 'filesystem')
-        output[:gauges].merge!(filesystem)
-      end
-      output
-    end
+option_parser.parse!
 
-    def self.filesystem
-      allocated, unused, max = `sysctl fs.file-nr`.split[-3..-1].map { |v| v.to_i }
-      open_files = allocated - unused
-      { 'filesystem.open_files' => open_files,
-        'filesystem.max_open_files' => max,
-        'filesystem.open_files_pct_max' => (open_files.to_f / max.to_f) * 100,
-      }
+if options[:user]
+  # Thanks Joe
```

Comment:
Durp, comment I wrote for myself to remember to thank the guy that wrote about [the correct way to drop privileges](http://timetobleed.com/5-things-you-dont-know-about-user-ids-that-will-destroy-you/). Will remove.